src/WebJobs.Script.WebHost/Controllers/InstanceController.cs

```diff
@@ -41,27 +41,23 @@ public async Task<IActionResult> Assign([FromBody] EncryptedHostAssignmentContex
 
             var assignmentContext = _startupContextProvider.SetContext(encryptedAssignmentContext);
 
-            // before starting the assignment we want to perform as much
-            // up front validation on the context as possible
-            string error = await _instanceManager.ValidateContext(assignmentContext);
-            if (error != null)
-            {
-                return StatusCode(StatusCodes.Status400BadRequest, error);
-            }
+            return await AssignInternal(assignmentContext);
+        }
 
-            // Wait for Sidecar specialization to complete before returning ok.
-            // This shouldn't take too long so ok to do this sequentially.
-            error = await _instanceManager.SpecializeMSISidecar(assignmentContext);
-            if (error != null)
+        [HttpPost]
+        [Route("admin/instance/assign2")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevel)]
+        public async Task<IActionResult> Assign2([FromBody] FunctionsWorkerContainerAssignmentContext workerAssignmentContext)
```

Comment:
Is this a host assignment or a worker assignment?

Reply:
Right - this is a HostAssignmentContext, right? I guess this points to a lack of tests in this PR - as you were writing the test you'd realize the mistake